### PR TITLE
[libc] adding linux SYS_fchmodat2 syscall.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libc/src/sys/stat/linux/chmod.cpp [hunk 1]
```diff
@@ -21,9 +21,12 @@ namespace LIBC_NAMESPACE {
...
   int ret = LIBC_NAMESPACE::syscall_impl<int>(SYS_chmod, path, mode);
+#elif defined(SYS_fchmodat2)
+  int ret =
+      LIBC_NAMESPACE::syscall_impl<int>(SYS_fchmodat2, AT_FDCWD, path, mode, 0, AT_SYMLINK_NOFOLLOW);
 #elif defined(SYS_fchmodat)
   int ret =
-      LIBC_NAMESPACE::syscall_impl<int>(SYS_fchmodat, AT_FDCWD, path, mode);
+      LIBC_NAMESPACE::syscall_impl<int>(SYS_fchmodat, AT_FDCWD, path, mode, 0);
 #else
 #error "chmod and fchmodat syscalls not available."
```
Comment (michaelrj-google) [inline libc/src/sys/stat/linux/chmod.cpp:31]: nit: add `fchmodat2` to this list

**Review by michaelrj-google — COMMENTED** [review]
this patch overall looks fine, I'll approve once you mark it not a draft, but it would be good to add something in the commit message about why this is being added.

**Review by michaelrj-google — APPROVED** [review]
LGTM, feel free to ping me to merge if you don't have access.

**Comment by devnexen** [comment]
No worries I m. member ;) thx for the review

**Pull request merged**